js-intro.js

```diff
@@ -6,39 +6,112 @@
 var mantra = "Be the dev"
 
 // 1a. Write the code that determines if there is a 'B' in mantra.
-
+// create a function that takes a string and turns it into an array
+// look for the index of the letter not equal to -1 to identify if it exists in the array 
+// return if it contains it or not
+
+const letterB = (variable) => {
+    let newArr = variable.split("")
+    for (let i=0; i < newArr.length; i++) {
+        if (newArr.indexOf('B') !== -1) {
+        return "This string contains a capital B"
+    } else {
+        return "This string does not contain a capital B"
+    }
+    }
+}
+console.log(letterB(mantra))
 
 
 // 1b. Write the code that determines if there is an 'x' in mantra.
-
+const letterX = (variable) => {
+    let newArr = variable.split("")
+    for (let i=0; i < newArr.length; i++) {
+        if (newArr.indexOf('x') !== -1) {
+        return "This string contains an x"
+    } else {
+        return "This string does not contain an x"
+    }
+    }
+}
+console.log(letterX(mantra))
 
 
 // 1c. Write the code that determines if there is a 'v' in mantra.
-
+const letterV = (variable) => {
+    let newArr = variable.split("")
+    for (let i=0; i < newArr.length; i++) {
+        if (newArr.indexOf('v') !== -1) {
+        return "This string contains a v"
+    } else {
+        return "This string does not contain a v"
+    }
+    }
+}
+console.log(letterV(mantra))
 
 
 // 1d. STRETCH: Write a function that can take two arguments - a string and a letter and returns a sentence telling the user if the letter exists within the string.
-
-
+var newMantra = "Happy Friday"
+var newLetter = "p"
+
+const letterCheck = (string, letter) => {
+    let newArr = string.split("")
+    for (let i=0; i < newArr.length; i++) {
+        if (newArr.indexOf(letter) !== -1) {
+        return `This string contains ${letter}`
+    } else {
+        return `This string does not contain ${letter}`
+    }
+    }
+}
+console.log(letterCheck(newMantra, newLetter))
 
 // ------------------------------ Consider the following variables:
 
 var myDog = "Snoopy"
 var myCat = "Garfield"
 
 // 2a. Create a function that returns the longer of the two variables. Expected output: "Garfield"
+// Create a function that takes two arguments, variables
+// splits the variables into 2 arrays
+// compares the length of each array
+// returns the longer word
 
+const biggerWord = (animalOne, animalTwo) => {
+    let arrayA = animalOne.split("")
+    let arrayB = animalTwo.split("")
+    if (arrayA.length > arrayB.length){
+        return `${animalOne}`
+    } else if (arrayA.length === arrayB.length) {
+        return `${animalOne} has the same amount of letters as ${animalTwo}.`
+    } else {
+        return `${animalTwo}`
+    }
+}
 
```

Comment:
You have totally correct logic here, but .length will work on strings as well as arrays so you can save yourself the step of splitting the string.